### PR TITLE
Fixes#7578: Updated/added quotes within error messages

### LINKED ISSUE #7578: Make usage of quotes in errors consistent
Our error messages are not consistent regarding the use of quotes, e.g. some quote the function name and contract name, others don't.

We should agree on how we want to do that and make it consistent everywhere.

@erak argues for double quotes `"` around all user defined names.
I tend to agree.

### PR DESCRIPTION
**Pull request by birtony**
### Description
Fixes #7578 

Updated existing quotes within error messages to double quotes, added new double quotes around user-defined variables.

### Checklist
- [ ] Code compiles correctly
- [ ] All tests are passing
- [ ] New tests have been created which fail without the change (if possible)
- [ ] README / documentation was extended, if necessary
- [ ] Changelog entry (if change is visible to the user)
- [ ] Used meaningful commit messages

### CODE REVIEW

**Comment by birtony** [comment]
@Marenz I have updated as many errors as I could find. Some tests fail due to the changes I have made to some of the error messages:
https://github.com/ethereum/solidity/blob/52ef581763c2a50b8cfb42953c702b7a033e7898/libsolidity/analysis/ViewPureChecker.cpp#L182
I have tried to update the tests accordingly:
https://github.com/ethereum/solidity/blob/52ef581763c2a50b8cfb42953c702b7a033e7898/test/libsolidity/syntaxTests/nameAndTypeResolution/010_type_conversion_for_comparison.sol#L6
https://github.com/ethereum/solidity/blob/52ef581763c2a50b8cfb42953c702b7a033e7898/test/libsolidity/syntaxTests/nameAndTypeResolution/011_type_conversion_for_comparison_invalid.sol#L5
https://github.com/ethereum/solidity/blob/52ef581763c2a50b8cfb42953c702b7a033e7898/test/libsolidity/syntaxTests/nameAndTypeResolution/013_large_string_literal.sol#L6
Please, let me know if there is anything that should update in my PR?

File: libsolidity/analysis/SyntaxChecker.cpp [hunk 1]
```diff
@@ -265,8 +265,8 @@ bool SyntaxChecker::visit(InlineAssembly const& _inlineAssembly)
 	if (yul::MSizeFinder::containsMSize(_inlineAssembly.dialect(), _inlineAssembly.operations()))
 		m_errorReporter.syntaxError(
 			_inlineAssembly.location(),
-			"The msize instruction cannot be used when the Yul optimizer is activated because "
-			"it can change its semantics. Either disable the Yul optimizer or do not use the instruction."
+			"The \"msize\" instruction cannot be used when the \"Yul\" optimizer is activated because "
+			"it can change its semantics. Either disable the \"Yul\" optimizer or do not use the instruction."
```
Comment (Marenz) [inline libsolidity/analysis/SyntaxChecker.cpp:269]: `Yul` is not a user defined name here :)

File: libsolidity/analysis/SyntaxChecker.cpp [hunk 2]
```diff
@@ -252,7 +252,7 @@ bool SyntaxChecker::visit(Literal const& _literal)
 bool SyntaxChecker::visit(UnaryOperation const& _operation)
 {
 	if (_operation.getOperator() == Token::Add)
-		m_errorReporter.syntaxError(_operation.location(), "Use of unary + is disallowed.");
+		m_errorReporter.syntaxError(_operation.location(), "Use of unary \"+\" is disallowed.");
```
Comment (Marenz) [inline libsolidity/analysis/SyntaxChecker.cpp:255]: I am unsure if we should quote `+`. Other opinions?

File: libsolidity/analysis/SyntaxChecker.cpp [hunk 3]
```diff
@@ -141,7 +141,7 @@ bool SyntaxChecker::visit(ModifierDefinition const&)
 void SyntaxChecker::endVisit(ModifierDefinition const& _modifier)
 {
 	if (!m_placeholderFound)
-		m_errorReporter.syntaxError(_modifier.body().location(), "Modifier body does not contain '_'.");
+		m_errorReporter.syntaxError(_modifier.body().location(), "Modifier body does not contain \"_\".");
```
Comment (Marenz) [inline libsolidity/analysis/SyntaxChecker.cpp:144]: Unsure if we should quote this. Other opinions?

File: test/libsolidity/syntaxTests/nameAndTypeResolution/010_type_conversion_for_comparison.sol [hunk 4]
```diff
@@ -3,4 +3,4 @@ contract test {
 }
 // ----
 // Warning: (42-63): Statement has no effect.
-// Warning: (20-66): Function state mutability can be restricted to pure
+// Warning: (20-66): Function state mutability can be restricted to "pure"
```
Comment (Marenz) [inline test/libsolidity/syntaxTests/nameAndTypeResolution/010_type_conversion_for_comparison.sol:6]: We have to check whether we want things like data types and visibility / pureness quoted

**Review by Marenz — CHANGES_REQUESTED** [review]
There still seem to be a lot of tests that need to be updated according to the CI.

**Comment by erak** [comment]
@Marenz Looking through the error messages, I'd propose the following rule of thumb for quotes:
- all keywords
- all types
- all user-defined types
- all opcodes
- all magic members
- (not sure yet) operators (like unary +)

@birtony There's a way to semi-automatically update the (at least .sol-file-based) tests. You just have to build and run `./build/test/tools/isoltest`, which compares the error messages stated by the file with the actual output from the compiler, and helps you overwrite the old error messages in the file via command line input.

**Comment by Marenz** [comment]
Your proposal sounds good to me @erak

**Comment by bshastry** [comment]
Needs rebase and test updates. IIUC, there seems to be consensus on writing a helper function to quote a string (see https://github.com/ethereum/solidity/issues/7578#issuecomment-563168826). IIUC, the helper function may look like so.

```
string quotedString(string s)
{
   return "\"" + s + "\"";
}
```

TBD: Placement of said function. My guess: somewhere in libdevcore.

**Comment by Marenz** [comment]
I'll take over this PR/feature

**Comment by Marenz** [comment]
@erak we often have types like `int_const` etc.. should we quote those as well? Or take this example:
```
Type int_const 5221...(1225 digits omitted)...5168 is not implicitly convertible to expected type int256.
```

**Comment by erak** [comment]
@Marenz I think it makes sense to also quote `int_const` since it's a type as well (even though an internal one). What's you take on this?

**Comment by Marenz** [comment]
I tend to go with quoting it as well

**Comment by Marenz** [comment]
Closing in favor of https://github.com/ethereum/solidity/pull/8103

**Pull request closed without merging**